app/src/main/java/com/github/se/travelpouch/ui/dashboard/Calendar.kt

```diff
@@ -0,0 +1,108 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import com.github.se.travelpouch.model.dashboard.ActivityRow
+import com.github.se.travelpouch.model.dashboard.CalendarView
+import com.github.se.travelpouch.model.dashboard.CalendarViewModel
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+/**
+ * Composable function to display the Calendar screen.
+ *
+ * @param calendarViewModel The ViewModel associated with the Calendar screen.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CalendarScreen(
+    calendarViewModel: CalendarViewModel,
+) {
+  // Observe the state of activities from the ViewModel
+  val calendarState by calendarViewModel.calendarState.collectAsState(initial = emptyList())
+
+  // Initial Setup
+  LaunchedEffect(Unit) {
+    // Set the default selected date to today
+    val today = Calendar.getInstance().time
+    calendarViewModel.onDateSelected(today)
+
+    calendarViewModel.activityViewModel.getAllActivities()
+  }
+
+  // Application
+  Scaffold(
+      topBar = {
+        // TopAppBar with title and icon to indicate the calendar feature
+        TopAppBar(
+            title = { Text("Calendar") },
+            navigationIcon = {
+              IconButton(
+                  modifier = Modifier.testTag("goBackIcon"),
+                  onClick = { /* TODO: Implement go back navigation logic */}) {
+                    // Back icon for navigation
+                    Icon(
+                        Icons.AutoMirrored.Filled.ArrowBack,
+                        contentDescription = null,
+                    )
+                  }
+            },
+            colors =
+                TopAppBarDefaults.topAppBarColors(
+                    containerColor = Color(0xFF6200EE), titleContentColor = Color.White),
+            modifier = Modifier.testTag("calendarTopAppBar"))
+      }) { innerPadding ->
+        Column(
+            modifier = Modifier.fillMaxSize().padding(innerPadding).testTag("calendarScreenColumn"),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Top) {
+              // CalendarView composable to wrap Android's CalendarView
+              CalendarView(
+                  selectedDate = calendarViewModel.selectedDate,
+                  onDateSelected = { date -> calendarViewModel.onDateSelected(date) },
+                  modifier = Modifier.fillMaxWidth().testTag("androidCalendarView"))
+
+              // Activities list
+              LazyColumn(modifier = Modifier.fillMaxSize().testTag("activityList")) {
+                val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
+                val selectedDateFormatted = dateFormat.format(calendarViewModel.selectedDate)
+                val selectedDayActivities =
+                    calendarState
+                        .filter { activity ->
```

Comment:
To align better with the MVVM model, it might be good to separate the filter from the UI and put in the view model a selectedDayActivities variable. What do you think ?